lib/node_modules/@stdlib/repl/lib/tokenizer.js

```diff
@@ -0,0 +1,329 @@
+			tokens.push( token );
+			return;
+		}
+		if ( token.type.binop || token.type.prefix || token.type.postfix || token.type.isAssign ) { // eslint-disable-line max-len
+			// Operator symbols - `+`, `=`, `++` etc:
+			token.type = 'operator';
+			tokens.push( token );
+			return;
+		}
+		if ( RE_PUNCTUATION.test( token.type.label ) ) {
+			// Punctuation symbols - `,`, `(`, `;` etc:
+			token.value = token.type.label;
+			token.type = 'punctuation';
+			tokens.push( token );
+		}
+	}
+
+	/**
+	* Callback invoked upon encountering a `Comment` when parsing.
+	*
+	* @private
+	* @param {Object} block - true if the comment is a block comment, false if it is a line comment
+	* @param {string} text - comment value
+	* @param {number} start - start index
+	* @param {number} end - end index
+	*/
+	function onComment( block, text, start, end ) {
+		if ( block ) {
+			// TODO: add support for highlighting multi-line comments.
+			return;
+		}
+		tokens.push({
+			'type': 'comment',
+			'start': start,
+			'end': end
+		});
+	}
+
+	/**
+	* Callback invoked upon encountering a `Identifier` AST node.
+	*
+	* @private
+	* @param {Object} node - AST node
+	*/
+	function Identifier( node ) {
+		var identifier;
+		var command;
+		var i;
+
+		// Ignore non-existent nodes:
+		if ( node.start === node.end ) {
+			return;
+		}
+		// If identifier is defined in the local scope, assume and treat it like a `variable` and push it as a token...
+		if ( contains( resolveLocalScope( ast, node ), node.name ) ) {
+			tokens.push({
+				'value': node.name,
+				'type': 'variable',
+				'start': node.start,
+				'end': node.end
+			});
+			return;
+		}
+		// If identifier is a REPL command, push it as a token...
+		for ( i = 0; i < COMMANDS.length; i++ ) {
+			command = COMMANDS[ i ];
+			if ( node.name === command[ 0 ] ) {
+				tokens.push( {
+					'value': node.name,
+					'type': 'command',
+					'start': node.start,
+					'end': node.end
+				});
+				return;
+			}
+		}
+		// If identifier is in global context, push it as a token...
+		identifier = context[ node.name ];
+		if ( identifier ) {
+			if ( isLiteralType( typeof identifier ) ) {
+				tokens.push( {
+					'value': node.name,
+					'type': 'variable',
+					'start': node.start,
+					'end': node.end
+				});
+			} else {
+				tokens.push( {
+					'value': node.name,
+					'type': typeof identifier,
+					'start': node.start,
+					'end': node.end
+				});
+			}
+		}
+	}
+
+	/**
+	* Callback invoked upon encountering a `MemberExpression` AST node.
+	*
+	* @private
+	* @param {Object} node - AST node
+	*/
+	function MemberExpression( node ) {
+		var properties = linkedList();
+		var property;
+		var locals;
+		var obj;
+
+		// Ignore non-existent nodes:
+		if ( node.start === node.end ) {
+			return;
+		}
+
+		/**
+		* Recursively traverses the `MemberExpression` node to populate the `properties` array.
+		*
+		* @private
+		* @param {Object} node - node to be traversed
+		*/
+		function traverse( node ) {
```

Comment:
Move this function declaration to the end of the function body, which is where we typically place nested function declarations.